design/design.md

```diff
@@ -0,0 +1,414 @@
+# ALBA implementation design document
+
+[@tolikzinovyev](https://github.com/tolikzinovyev) November 2024
+
+## Introduction
+
+This is the first design document for our Rust implementation of
+Approximate Lower Bound Arguments (ALBA) intended to guide the development
+of the library.
+We aim to create a high quality reference implementation of ALBA sufficient
+for accurately prototyping larger protocols utilizing this primitive.
+Applications that could benefit, at least in the Cardano universe, include
+Mithril, Peras, Leios and cross-chain bridges.
+These use cases will generally guide the development of our library, though
+our aim is to also allow the open-source community to study and experiment
+with ALBA.
+
+### Overview of ALBA
```

Comment:
This is way too long, we already have a README and doc that you can link.
You should write the overall concept quickly and refer to them.

Reply:
This document is about the design of our implementation and the choices we have made. Someone reading this should already know what Alba is (or seen the README on the first page).